src/convert/declarations.test.ts

```diff
@@ -28,15 +28,39 @@ describe("transform declarations", () => {
       expect(await transform(src)).toBe(expected);
     });
 
-    it("transforms typeof imports", async () => {
-      const src = `import typeof {foo} from './foo';`;
-      const expected = `import {foo} from './foo';`;
-      expect(await transform(src)).toBe(expected);
-    });
-    it("transforms default typeof imports", async () => {
-      const src = `import typeof Foo from './foo';`;
-      const expected = `import Foo from './foo';`;
-      expect(await transform(src)).toBe(expected);
+    describe("import typeof", () => {
+      it("import typeof Foo from './foo'", async () => {
+        const src = `import typeof Foo from './foo'`;
+
+        expect(await transform(src)).toMatchInlineSnapshot(
+          `"type Foo = typeof import('./foo');"`
```

Comment:
Err... wait what? That's really confusing. Why would we do an async import?

It's unclear to me what you mean about bundling. If we import something from a file and it's only used for types, then it'll get stripped out after typescript has processed the file, correct?

Reply:
The problem is that TS doesn't support `typeof` imports which means that we'd have to do:
```
import Foo from "./foo";

class Bar {
   Foo: typeof Foo | null,
   componentDidMount() {
       import("./foo").then((m) => this.Foo = m);
   }
}
```
I'm not sure if our build system will strip out the `import Foo from "./foo"` or not.  Even if it does, it would be really easy for someone to accidentally use `Foo` as a value within the file which would nullify the value of the deferred import.